### ISSUE TITLE
Set up an org

### ISSUE THREAD

**Issue by Gaurav0** [sanderfrenken/Universal-LPC-Spritesheet-Character-Generator#231]
I wasn't able to figure out how to transfer this to you, so I set up an org, and invited you as an owner.

**Comment by jrconway3** [comment]
Heh all right. Sounds good. Perhaps this will make things less confusing so we don't have to keep opening PR's to your base branch by mistake.

**Comment by Gaurav0** [comment]
Just updated LiberatedPixelCup/master. Please note that if a PR is accepted on this repo, that repo is not updated automatically.

**Comment by sanderfrenken** [comment]
Hi! what is the purpose of this org exactly? I am not familiar with orgs and what in this case is the benefit?

**Comment by Gaurav0** [comment]
It's not letting me transfer my repo to you because "you already have a repo with the same name". So I created an org since that would basically accomplish the same thing: give you admin access to my repo.

The other benefit is that we can all act as admins and/or give others commit access (or fine tuned access). This can be a benefit for quicker turnaround on PRs and/or work on setting up. Having @jrconway3 be able to review art related PRs might be a benefit, for example.

**Comment by jrconway3** [comment]
I've been in orgs before. The main purpose is to allow multiple people to be able to manage a repo. But as Guarav said, its really to get the root repository into a location where its more flexible.

Right now both Sanderfrenken and I frequently update our own forks, and bluecarrot16 occasionally drops updates as well. There's tons and tons of forks right now and its difficult to select the right one when trying to open a PR. Usually I come here to this fork right after pushing updates and click on the banner that said "recently pushed branch" and open a PR using that button instead.

This org can actually be pretty useful, though. If in the future Sanderfrenken stops maintaining it, it could be passed onto someone else instead.


However, we'd have to keep up with the root repository again.

**Comment by jrconway3** [comment]
So because I was given admin permissions on the root I did some small cleanup. I removed the very obvious outdated issues (golden helmets missing down for example) as well as two outdated PR's. But I'm trying to leave open stuff that's more questionable. (The third PR might have a useful tip; we can't use said PR because its way outdated, but potentially the PR might contain some fixes we could still implement.)



I guess I should add a root remote to my local clone as well to point towards the org's repo.

**Comment by sanderfrenken** [comment]
ahh ok thanks for clarifying. In that case, wouldn't it make more sense to just work on the org repo and leave this one for what it is?

I closed two open issues in the org repo as well, didn't see you already looked at it. But these issues were also outdated IMO.

**Comment by jrconway3** [comment]
Yeah, I think that's the idea, but its your call. You've been managing this for a long time now.


What you could do is keep using this fork for when you make changes and make updates, while using the org repo as the primary one, which of course is Guarav's original now updated to your fork and on the "organization". So now we can all update back to the original one instead, which also makes opening PR's a lot easier.


The only issue though would be the generator itself, which is currently on your .io:
https://sanderfrenken.github.io/Universal-LPC-Spritesheet-Character-Generator/

IDK if there's a way to forward it to the org's name instead:
https://liberatedpixelcup.github.io/Universal-LPC-Spritesheet-Character-Generator/


What I don't want to do here is remove the URL entirely, because so many rely on that link now. If we could forward the link that might work.

**Comment by jrconway3** [comment]
> I closed two open issues in the org repo as well, didn't see you already looked at it. But these issues were also outdated IMO.

I looked at all the issues, but these two issues in particular I wasn't exactly sure on, so I didn't want to impose lol. I knew there was an export option here, but I wasn't sure if the export option was accurate.


Speaking of which, the export option is a bit of a problem I overlooked before when doing some modifications. The ability to have presets is a big thing, and changing asset names ruins that. Granted, the only reason that usually happens is when similar assets are introduced, it can become difficult to figure out the major differences between them... I think its more justifiable when the original assets weren't actually named, or in situations where the name is so generic that adding a bit more detail is fine. Still, though, its something I need to be careful of in the future.

**Comment by sanderfrenken** [comment]
> Yeah, I think that's the idea, but its your call. You've been managing this for a long time now.
> 
> What you could do is keep using this fork for when you make changes and make updates, while using the org repo as the primary one, which of course is Guarav's original now updated to your fork and on the "organization". So now we can all update back to the original one instead, which also makes opening PR's a lot easier.
> 
> The only issue though would be the generator itself, which is currently on your .io: https://sanderfrenken.github.io/Universal-LPC-Spritesheet-Character-Generator/
> 
> IDK if there's a way to forward it to the org's name instead: https://liberatedpixelcup.github.io/Universal-LPC-Spritesheet-Character-Generator/
> 
> What I don't want to do here is remove the URL entirely, because so many rely on that link now. If we could forward the link that might work.

I think it makes sense to use the other now :) No hard feelings from my side at least.

But indeed that GitHub.io page needs some thought then. Let's continue for now on my repo, and when we can work it out with the domain in a nice way let's switch.

An easy but ugly fix would be in my fork to alter the index.html to show a text "We moved!" with a button to open the org GitHub pages instead. It would render my fork useless but I usually don't work with forks anyway, I can just branch of the org repo instead.

**Comment by sanderfrenken** [comment]
> > I closed two open issues in the org repo as well, didn't see you already looked at it. But these issues were also outdated IMO.
> 
> I looked at all the issues, but these two issues in particular I wasn't exactly sure on, so I didn't want to impose lol. I knew there was an export option here, but I wasn't sure if the export option was accurate.
> 
> Speaking of which, the export option is a bit of a problem I overlooked before when doing some modifications. The ability to have presets is a big thing, and changing asset names ruins that. Granted, the only reason that usually happens is when similar assets are introduced, it can become difficult to figure out the major differences between them... I think its more justifiable when the original assets weren't actually named, or in situations where the name is so generic that adding a bit more detail is fine. Still, though, its something I need to be careful of in the future.

with presets, you mean people using the URL to recreate a previously created sheet right? And indeed that will break if you have a URL that uses an asset name inside that was changed later. 

Agree that is something to try to avoid. However, I have no idea how many people would do this anyway, use presets. I personally see the generator really as a tool for gamedev, and never found the urge to use presets. But maybe others do. 

If we would see more request for a system of saving and loading presets, we could think of a more structural solution instead of relying on asset names. We might need to work with some form of ID's for example, but that would require quite some work.

**Comment by jrconway3** [comment]
Yeah, using the URL, but not just that, I think that card you closed with the presets issue was asking about a preset option more than simply using the URL. I know there's an export/import option, in this case using json, so that solves the issue. That's basically what I meant--any form of import or preset functionality (so either using the URL or using a json import).

That being said, I can say it doesn't "break" if you use an asset that's name was changed. All that happens is the asset doesn't load in anymore, but it doesn't break the page or anything.

**Comment by jrconway3** [comment]
> An easy but ugly fix would be in my fork to alter the index.html to show a text "We moved!" with a button to open the org GitHub pages instead. It would render my fork useless but I usually don't work with forks anyway, I can just branch of the org repo instead.

Nah we don't have to do that at all. We an easily drop in a quick redirect instead. If you don't intend on updating your fork anymore, you can easily just include a forward in .js that redirects to the new page. No need to have a "We moved!" page. Over time people will get used to using the new URL instead of the old one.

**Comment by sanderfrenken** [comment]
> Yeah, using the URL, but not just that, I think that card you closed with the presets issue was asking about a preset option more than simply using the URL. I know there's an export/import option, in this case using json, so that solves the issue. That's basically what I meant--any form of import or preset functionality (so either using the URL or using a json import).
> 
> That being said, I can say it doesn't "break" if you use an asset that's name was changed. All that happens is the asset doesn't load in anymore, but it doesn't break the page or anything.

yes indeed, and no it doesn't break. But like functionally it does break for the user (as the asset is not restored properly).

**Comment by sanderfrenken** [comment]
> > An easy but ugly fix would be in my fork to alter the index.html to show a text "We moved!" with a button to open the org GitHub pages instead. It would render my fork useless but I usually don't work with forks anyway, I can just branch of the org repo instead.
> 
> Nah we don't have to do that at all. We an easily drop in a quick redirect instead. If you don't intend on updating your fork anymore, you can easily just include a forward in .js that redirects to the new page. No need to have a "We moved!" page. Over time people will get used to using the new URL instead of the old one.

ah well that would be perfect then. Then let's merge the credits PR when you looked at it, and sync it to the org. Then we can land a PR here to do the redirect and we are good to go. It sounds as if you now how to handle the redirect, would you be able to create a PR for that?

I can adjust this projects readme to refer to the new repo.

**Comment by jrconway3** [comment]
Right, well, while we could do that, it would be more ideal to do a proper 301 redirect. I don't know how github.io pages work, so I thought maybe there was a way to set it up to just forward the entire page instead of dropping it into the repo fork itself. That would be more ideal I think.

I'll do some research and see if I can figure that out.

**Comment by jrconway3** [comment]
This is not what I was expecting:
https://gist.github.com/domenic/1f286d415559b56d725bee51a62c24a7


But apparently, you can create a redirect by just creating a new repository and setting that repository as your github page, so it doesn't have to affect your fork. However, you still need to have a repo set up, and if we do this, we would break all preset links of course. Its possible we could implement a js-method to handle keeping any existing preset links for the few users that would actually want that functionality. Granted, we could still do that even with a new repository, too...


The way they handle html redirects is this:
```
<meta http-equiv="refresh" content="0; URL=https://bob.github.io/repo/">
```
I wasn't aware this was a thing, but apparently HTML has a built-in method to handle redirects now.

**Comment by jrconway3** [comment]
301 redirects are out of the question for Github Pages, so that solves that dilemma. Okay, the best method is using the HTML redirect instead.

**Comment by sanderfrenken** [comment]
> This is not what I was expecting: https://gist.github.com/domenic/1f286d415559b56d725bee51a62c24a7
> 
> But apparently, you can create a redirect by just creating a new repository and setting that repository as your github page, so it doesn't have to affect your fork. However, you still need to have a repo set up, and if we do this, we would break all preset links of course. Its possible we could implement a js-method to handle keeping any existing preset links for the few users that would actually want that functionality. Granted, we could still do that even with a new repository, too...
> 
> The way they handle html redirects is this:
> 
> ```
> <meta http-equiv="refresh" content="0; URL=https://bob.github.io/repo/">
> ```
> 
> I wasn't aware this was a thing, but apparently HTML has a built-in method to handle redirects now.

oh wow that works very easy indeed. Preset links we could fix in the HTML redirect maybe? By taking the query params from the current URL and append them to the redirect target?

**Comment by Gaurav0** [comment]
If you want to preserve the hash (url fragment that mimics query params) only a javascript redirect will work. Fragment urls are not even sent to the server.

**Comment by Gaurav0** [comment]
Another option is to keep both repos in sync automatically using a github action

**Comment by sanderfrenken** [comment]
> If you want to preserve the hash (url fragment that mimics query params) only a javascript redirect will work. Fragment urls are not even sent to the server.

ah ok wasnt aware of that, ok then those will break. I think its acceptable though, one can easily fix their presets by a simple find+replace right.

**Comment by sanderfrenken** [comment]
> Another option is to keep both repos in sync automatically using a github action

Hmm if the only argument to do that (syncing and keep using my fork) is the presets that might break for some, I would say let's not opt for that route. Agree?

**Comment by Gaurav0** [comment]
I think since you have been maintaining this it is really up to you.

**Comment by jrconway3** [comment]
Let me look into a javascript redirect first. It shouldn't be too hard to add that in.

**Comment by sanderfrenken** [comment]
> I think since you have been maintaining this it is really up to you.

That is kind of you to say. I really think it's a good idea, it's not mine anyway for a large part and I want all of you to easily work with it as well. Plus, if something happens to me, then at least the org repo will remain properly usable for everyone :)

**Comment by jrconway3** [comment]
Got it. I tested it first by using my local, then set it to the live instance so it works only if its on sanderfrenken.github.io. This allows it to still work on local / custom environments so we don't have to worry about this fork not being able to be used anymore.

Here's it in action:
https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/compare/license-handling...jrconway3:Universal-LPC-Spritesheet-Character-Generator:issue-231-forward-to-org?expand=1

I created a new lightweight js file that only handles the redirect, and have it as the first entry in source_index.html. The reason is so this triggers first before anything else to make it redirect ASAP if it detects a redirect needs to happen. The redirect happens if the hostname matches `from`:

```
const from = 'sanderfrenken.github.io'
```


(I have it going to license-handling because I assume that's being merged tomorrow, so that reduces any conflicts that might happen.)

**Comment by sanderfrenken** [comment]
Ok it's done! We can/ should now use the org repo instead of this fork.
I will update the org repo README to show the correct links etc.

We also need to move the issues from here to there.

**Comment by sanderfrenken** [comment]
> Got it. I tested it first by using my local, then set it to the live instance so it works only if its on sanderfrenken.github.io. This allows it to still work on local / custom environments so we don't have to worry about this fork not being able to be used anymore.
> 
> Here's it in action: https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/compare/license-handling...jrconway3:Universal-LPC-Spritesheet-Character-Generator:issue-231-forward-to-org?expand=1
> 
> I created a new lightweight js file that only handles the redirect, and have it as the first entry in source_index.html. The reason is so this triggers first before anything else to make it redirect ASAP if it detects a redirect needs to happen. The redirect happens if the hostname matches `from`:
> 
> ```
> const from = 'sanderfrenken.github.io'
> ```
> 
> (I have it going to license-handling because I assume that's being merged tomorrow, so that reduces any conflicts that might happen.)

thanks for this btw. works like a charm :)

**Comment by sanderfrenken** [comment]
I am not able to use the transfer issue functionality in order to transfer the 2 remaining issues to the new repo.

Not sure how valid they still are/ what we need to address still. @jrconway3  can you check what to do with the remaining 2 issues?

**Comment by jrconway3** [comment]
I'll reopen them over there but clean up the comments with new details. I really want that idle/sit/etc one to be there just so there's a universal "please don't create cards about this" issue open.

I think we should also create one more card here--that being a "don't open new cards, we've moved, go here instead" issue.

**Comment by jrconway3** [comment]
I created a READ FIRST issue in here that I think we can leave behind forwarding people to the new repo and issues list. Just in case people come into the issues list and don't see the updated readme.

**Comment by jrconway3** [comment]
There is one small flaw with the transfer, but it shouldn't be a big deal. The problem I mentioned before about the licenses could be an issue here--that being getting rid of the credits.csv. In this case we're not getting rid of it, though, its just not going to be updated anymore, so old credits linking to it should still apply. As such, we shouldn't have a problem here. Just a minor concern I only thought of now.

**Comment by Gaurav0** [comment]
Anyone else we should invite to the org? And with what level of permissions?

In my experience, we want as many people to be involved as possible.

**Comment by Gaurav0** [comment]
@sanderfrenken Are you in the OGA discord?

**Comment by sanderfrenken** [comment]
> There is one small flaw with the transfer, but it shouldn't be a big deal. The problem I mentioned before about the licenses could be an issue here--that being getting rid of the credits.csv. In this case we're not getting rid of it, though, its just not going to be updated anymore, so old credits linking to it should still apply. As such, we shouldn't have a problem here. Just a minor concern I only thought of now.

Yes I thought about that. That is why for now kept the old credits.csv in there. For now, if there are new submissions (which is very unlikely to happen) in the coming days, we need to pay attention to them being added to credits.csv (only temporarily) and most importantly, have valid credits entries in the json file (this we will have to be checking forever from now on ofc).


I will fix the credits.csv generation somewhere tomorrow or the day after, so the temporary stuff will be no issue.

**Comment by sanderfrenken** [comment]
anyways, thanks a lot @Gaurav0 and @jrconway3 , I think this is very nice team work and good progress. Moreover, thanks for transferring the repo @Gaurav0

**Comment by sanderfrenken** [comment]
> Anyone else we should invite to the org? And with what level of permissions?
> 
> In my experience, we want as many people to be involved as possible.

I would say bluecarrot and ben creating out of my head at least. I can do that now.

**Comment by sanderfrenken** [comment]
> @sanderfrenken Are you in the OGA discord?

ow no didn't know of its existence either

**Comment by sanderfrenken** [comment]
> > Anyone else we should invite to the org? And with what level of permissions?
> > In my experience, we want as many people to be involved as possible.
> 
> I would say bluecarrot and ben creating out of my head at least. I can do that now.

invited them

**Comment by Gaurav0** [comment]
We should make an announcement on OGA forums and on OGA discord.

**Comment by jrconway3** [comment]
I haven't seen Ben Creating do much in a long time and ElizaWy is primarily doing her own thing. I don't think she has contributed to the repo directly that much.

The big one is definitely Bluecarrot16 as he's the only other one fairly actively contributing to the repo. Much of the work done in the repo since I left to begin with has been Bluecarrot16. He did some stuff early last year and a few weeks ago, but I think he was mostly absent last year. Basically, the same time I've been back (I came back around April), bluecarrot16 has been inactive, only becoming active again in the last couple months. (On OGA, he posted the backpacks update in March, then the next update after that was November.)


I almost shared the org in the OGA Discord but opted not to for now. I know Gaurav is in it because he started DM'ing me and it says we're both in the discord.

**Comment by jrconway3** [comment]
If they're willing to join that would be great, but yeah, we should see if anything else wants to join as well.

**Comment by Gaurav0** [comment]
Generally speaking, we should offer an invitation of write level permission to pretty much anyone who has significantly contributed. That way we build a sense of community. if they try to mess things up we can always ban them and fix things easily.

We can also hand out moderator level of permission to anyone who volunteers to moderate.

There is no point to handing out "read" level of permission in a public repo, everyone not blocked/banned has that anyway.

Who gets admin level of permission I defer to whoever @sanderfrenken trusts. They can do a lot of damage.

**Comment by sanderfrenken** [comment]
> I haven't seen Ben Creating do much in a long time and ElizaWy is primarily doing her own thing. I don't think she has contributed to the repo directly that much.
> 
> The big one is definitely Bluecarrot16 as he's the only other one fairly actively contributing to the repo. Much of the work done in the repo since I left to begin with has been Bluecarrot16. He did some stuff early last year and a few weeks ago, but I think he was mostly absent last year. Basically, the same time I've been back (I came back around April), bluecarrot16 has been inactive, only becoming active again in the last couple months. (On OGA, he posted the backpacks update in March, then the next update after that was November.)
> 
> I almost shared the org in the OGA Discord but opted not to for now. I know Gaurav is in it because he started DM'ing me and it says we're both in the discord.

I added both Ben and Bluecarrot, and explained a bit our new situation. I speak to them so now and then personally as they do some commissions, and hopefully they accept the request. Ben started a new take on the generator a while back (2 years already, time flies), but he also knows very well how the current generator works.

Anyways, I added them as owners to the org itself. I figured that is the same level as I have, and I trust them with that. But if it should be done in another way let me know.

Refrained from adding Eliza, she is indeed not really occupied with the generator stuff anyways, more about the art itself.

**Comment by sanderfrenken** [comment]
> Generally speaking, we should offer an invitation of write level permission to pretty much anyone who has significantly contributed. That way we build a sense of community. if they try to mess things up we can always ban them and fix things easily.
> 
> We can also hand out moderator level of permission to anyone who volunteers to moderate.
> 
> There is no point to handing out "read" level of permission in a public repo, everyone not blocked/banned has that anyway.
> 
> Who gets admin level of permission I defer to whoever @sanderfrenken trusts. They can do a lot of damage.

Maybe spread the word in the OGA discord indeed? I would say the more the merrier with regard to write permission. I think our master branch is protected right? So no-one will be able to one his/ her own merge stuff to master.

How can I get into the OGA discord? I could join if allowed.

**Comment by jrconway3** [comment]
Yeah as I said, Bluecarrot16 has probably been the biggest contributor in recent years and has been the one most active in keeping this repo updated. This past year I have been the biggest contributor, but Bluecarrot16 has probably done far more than I did overall.

On OGA there's a fair decent number of contributors as well, but IDK if they contributed to the repo directly.


The Discord link is in the upper left corner on OGA itself:
https://opengameart.org/

Or here for a direct link:
https://discord.gg/yDaQ4NcCux

**Comment by jrconway3** [comment]
So I went ahead and dropped a comment on the discord server now, but it probably would be a good idea to create a forum thread as well.

**Comment by sanderfrenken** [comment]
> So I went ahead and dropped a comment on the discord server now, but it probably would be a good idea to create a forum thread as well.

What would you think about creating an LPC dedicated discord server?

**Comment by jrconway3** [comment]
I don't see much of a point in doing that atm. Maybe if there's enough people who want to join, but for now the OGA discord server and the Discussions threads (https://github.com/LiberatedPixelCup/Universal-LPC-Spritesheet-Character-Generator/discussions) should be fine I think.


LPC is OGA's project anyway so it makes sense to remain associated with OGA. I don't see a need to create a dedicated server unless we get a lot of people who ONLY want to talk about LPC, and even then, we can always just ask MedicineStorm for a dedicated LPC channel in the OGA discord. I'm sure she'd be happy to oblige.

**Comment by sanderfrenken** [comment]
Alright, that makes sense. Then let's stick with OGA :)
